src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -53,31 +53,53 @@
        :changed-path (.. watched-path (resolve (.context event)) (toFile))})))
 
 (defn validate-watch-options!
+  "Validate that the options supplied include a valid Boolean value for key :recursive."
   [options]
-  (when-not (= true (:recursive options))
+  (when-not (instance? Boolean (:recursive options))
     (throw
       (IllegalArgumentException.
-        (trs "Support for non-recursive directory watching not yet implemented")))))
+        (trs "Must pass a boolean value for :recursive (directory watching) option")))))
 
 (defrecord WatcherImpl
-  [watch-service callbacks]
+  [watch-service callbacks recursive]
   Watcher
   (add-watch-dir!
+    [this dir]
+    (let [watched-path (.toPath (fs/file dir))]
+      (if @recursive
+        (DirWatchUtils/registerRecursive watch-service [watched-path])
+        (DirWatchUtils/register watch-service watched-path))))
+
+  (add-watch-dir!
     [this dir options]
     (validate-watch-options! options)
```

Comment:
Do you know if we will always automatically emit that deprecation because of our use for HA? @haus  can we update HA to use the new form?